### ISSUE TITLE
Add support for AKS agent pools

### ISSUE THREAD

**Issue by jbw976** [crossplane/provider-azure#262]
### What problem are you facing?

AKS clusters currently have no way via Crossplane to use multiple node pools.   The docs for provider-azure show some fields for number of nodes and node size, but not for setting more details for the “agent pool” or declaring multiple of them: https://doc.crds.dev/github.com/crossplane/provider-azure/compute.azure.crossplane.io/AKSCluster/v1alpha3@v0.16.1 

### How could Crossplane help solve your problem?

The `AKSCluster` type could expose fields to configure and manage a set of agent pools for the cluster.  This article describes some of the operational steps for reference: https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools

**Comment by bergerx** [comment]
> The AKSCluster type could expose fields to configure and manage a set of agent pools for the cluster.

This statement could be misleading, and implementing the solution this way would be against how the crossplane aims to manage external resources. Even though some nodepool details are needed to create an AKS Cluster, nodepool is a different managed resource itself.

Here is a quote from [the related thread in the slack](https://crossplane.slack.com/archives/C01718T2476/p1622997552064100?thread_ts=1622983455.063700&cid=C01718T2476):
> I would expect the kind name for the managed resource to follow the azure go SDK's terminology. I find the "AKSCluster" name a bit weird in this respect (azure SDK for go uses `ManagedCluster` for **AKSclusters** and `AgentPool` for **nodepools** following the Azure Rest API).

**Comment by jbw976** [comment]
Good point @bergerx, thanks for calling that out! I've updated the description to make that more clear :)
